src/gt4py/next/otf/binding/pybind.py

```diff
@@ -132,59 +132,39 @@ def visit_FunctionCall(self, call: FunctionCall):
         args = [self.visit(arg) for arg in call.args]
         return cpp_interface.render_function_call(call.target, args)
 
-    def visit_SidConversion(self, sid: SidConversion):
+    def visit_BufferSID(self, sid: BufferSID):
         return self.generic_visit(
-            sid, rendered_scalar_type=cpp_interface.render_python_type(sid.scalar_type.type)
+            sid, rendered_scalar_type=cpp_interface.render_scalar_type(sid.scalar_type)
         )
 
-    SidConversion = as_jinja(
+    BufferSID = as_jinja(
         """gridtools::sid::rename_numbered_dimensions<{{", ".join(dimensions)}}>(
                 gridtools::as_sid<{{rendered_scalar_type}},\
                                   {{dimensions.__len__()}},\
                                   gridtools::integral_constant<int, {{dim_config}}>,\
-                                  999'999'999>({{buffer_name}})
+                                  999'999'999>({{source_buffer}})
             )"""
     )
 
     DimensionType = as_jinja("""generated::{{name}}_t""")
 
 
 def make_parameter(
```

Comment:
Not important, but is this maker function really needed? If it is only used in this module, I would say it only hides the mapping from an `interface.Parameter` to a `FunctionParameter`

Reply:
inlined